From 402151d0e07bbe1ea64f6cc9f13a1098a8cbe804 Mon Sep 17 00:00:00 2001
From: M Z <md@any.ccc>
Date: Mon, 30 Jun 2025 18:19:04 -0400
Subject: [PATCH] kmod

---
 include/rootfs.mk | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/include/rootfs.mk b/include/rootfs.mk
index 26f249d818..f9d56ed3cb 100644
--- a/include/rootfs.mk
+++ b/include/rootfs.mk
@@ -74,6 +74,7 @@ define prepare_rootfs
 	fi)
 	@mkdir -p $(1)/etc/rc.d
 	@mkdir -p $(1)/var/lock
+	@mkdir -p $(1)/www/repo
 	@( \
 		cd $(1); \
 		if [ -n "$(CONFIG_USE_APK)" ]; then \
@@ -119,6 +120,11 @@ define prepare_rootfs
 		$(1)/usr/lib/opkg/info/*.postinst* \
 		$(1)/usr/lib/opkg/lists/* \
 		$(1)/var/lock/*.lock
+	@(      $(MAKE) package/index; \
+	        $(CP) $(TOPDIR)/bin/targets/$(BOARD)/$(SUBTARGET)/packages $(1)/www/repo/packages; \
+	        $(SED) "/openwrt_core/c\src/gz openwrt_core file:///www/repo/packages" $(1)/etc/opkg/distfeeds.conf; \
+		$(SED) "/openwrt_kmods/c\## src/gz openwrt_kmods " $(1)/etc/opkg/distfeeds.conf; \
+	)
 	$(call clean_ipkg,$(1))
 	$(call mklibs,$(1))
 	$(if $(SOURCE_DATE_EPOCH),find $(1)/ -mindepth 1 -execdir touch -hcd "@$(SOURCE_DATE_EPOCH)" "{}" +)
-- 
2.48.1

